library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -998,6 +1044,34 @@ public void setMaxRetryCount(int maxRetryCount)
   }
 
   /**
+   * This class scans a directory and put the list of files in a blocking queue.
+   */
+  public class DirectoryScannerThread implements Runnable
+  {
+    @Override
+    public void run()
+    {
+      try {
+        while (execute) {
+          if (pendingFiles.isEmpty() && fileQueue.isEmpty() && (System.currentTimeMillis() - scanIntervalMillis >= lastScanMillis)) {
+            LinkedHashSet<Path> newFileSet = scanner.scan(fs, filePath, processedFiles);
+            for (Path newFile : newFileSet) {
+              fileQueue.put(newFile);
```

Comment:
does not help to put limit on the size of fileQueue, as newFileSet already has all the file name, And emitTuple is scanning  only max 1024 entries per second.